streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1096,7 +1096,7 @@ private Optional<String> removeStreamThread(final long timeoutMs) throws Timeout
                 // make a copy of threads to avoid holding lock
                 for (final StreamThread streamThread : new ArrayList<>(threads)) {
                     final boolean callingThreadIsNotCurrentStreamThread = !streamThread.getName().equals(Thread.currentThread().getName());
-                    if (streamThread.isAlive() && (callingThreadIsNotCurrentStreamThread || getNumLiveStreamThreads() == 1)) {
```

Comment:
Note that `streamThread.isAlive()` is a `native` method and Mockito currently does not support mocking of `native` methods. 

I have made this change in logic to remove the check because even if the thread is not running, we should remove it from data structures such as `threads` and perform the clean up of the Java object. Hence, the check is not really necessary.

@cadonna let me know if you have better ideas here.

Reply:
This is resolved. See the conversation at https://github.com/apache/kafka/pull/12465#discussion_r984479081